### PR TITLE
Build NixOS kernel with SEV support

### PR DESCRIPTION
**Pull request by connorkuehl**
Use the boot.kernelPatches option to enable the requisite config change
for SEV support.

Reference: https://nixos.wiki/wiki/Linux_kernel#Custom_configuration

### CODE REVIEW

**Review by npmccallum — CHANGES_REQUESTED** [review]
https://nixos.org/nixos/manual/#sec-kernel-config

Please use the technique outlined there.

**Comment by connorkuehl** [comment]
> https://nixos.org/nixos/manual/#sec-kernel-config
> 
> Please use the technique outlined there.

Just did something similar described here: https://nixos.wiki/wiki/Linux_kernel#Custom_configuration -- I thought this might be preferable because it didn't require use to specify specifically which kernel package we want to use and override with the option; it will just add the config option to whichever kernel we happen to decide to use via the package manager.

File: configuration.nix [hunk 1]
```diff
@@ -25,6 +25,13 @@ in {
 
   boot.loader.systemd-boot.enable = true;
   boot.loader.efi.canTouchEfiVariables = true;
+  boot.kernelPatches = [{
+    name = "kvm-amd-sev";
+    patch = null;
+    extraConfig = ''
+          KVM_AMD_SEV y
```
Comment (npmccallum) [inline configuration.nix]:
```suggestion
          CRYPTO_DEV_CCP y
          CRYPTO_DEV_CCP_DD m
          CRYPTO_DEV_SP_PSP y
          KVM_AMD_SEV y
```

**Review by npmccallum — CHANGES_REQUESTED** [review]

**Review by npmccallum — CHANGES_REQUESTED** [review]
Almost working! We need a udev rule. Currently:

```sh
$ ls -lh /dev/sev
crw------- 1 root root 10, 58 Jan 31 15:00 /dev/sev
```

**Review by npmccallum — APPROVED** [review]

**Pull request merged**